### DISCUSSION TITLE
Docker Prune Error

### DISCUSSION THREAD

**Discussion by stondino00** [Security-Onion-Solutions/securityonion#4034] [Unsupported Versions]
Upon running sudo salt-call state.highstate I get one single fail at the end.

How do I manually prune this old 2.3.30 image?

----------
          ID: prune_images
    Function: cmd.run
        Name: so-docker-prune
      Result: False
     Comment: Command "so-docker-prune" run
     Started: 17:54:47.810316
    Duration: 2497.444 ms
     Changes:
              ----------
              pid:
                  24047
              retcode:
                  1
              stderr:
                  Traceback (most recent call last):
                    File "/usr/lib/python3.6/site-packages/docker/api/client.py", line 222, in _raise_for_status
                      response.raise_for_status()
                    File "/usr/lib/python3.6/site-packages/requests/models.py", line 928, in raise_for_status
                      raise HTTPError(http_error_msg, response=self)
                  requests.exceptions.HTTPError: 409 Client Error: Conflict for url: http+docker://localunixsocket/v1.30/images/ghcr.io/security-onion-solutions/so-curator:2.3.30?force=False&noprune=False

                  During handling of the above exception, another exception occurred:

                  Traceback (most recent call last):
                    File "/sbin/so-docker-prune", line 89, in <module>
                      main(args.quiet)
                    File "/sbin/so-docker-prune", line 75, in main
                      client.images.remove(tag)
                    File "/usr/lib/python3.6/site-packages/docker/models/images.py", line 288, in remove
                      self.client.api.remove_image(*args, **kwargs)
                    File "/usr/lib/python3.6/site-packages/docker/utils/decorators.py", line 19, in wrapped
                      return f(self, resource_id, *args, **kwargs)
                    File "/usr/lib/python3.6/site-packages/docker/api/image.py", line 481, in remove_image
                      return self._result(res, True)
                    File "/usr/lib/python3.6/site-packages/docker/api/client.py", line 228, in _result
                      self._raise_for_status(response)
                    File "/usr/lib/python3.6/site-packages/docker/api/client.py", line 224, in _raise_for_status
                      raise create_api_error_from_http_exception(e)
                    File "/usr/lib/python3.6/site-packages/docker/errors.py", line 31, in create_api_error_from_http_exception
                      raise cls(e, response=response, explanation=explanation)
                  docker.errors.APIError: 409 Client Error: Conflict ("conflict: unable to remove repository reference "ghcr.io/security-onion-solutions/so-curator:2.3.30" (must force) - container 28b45b6497c7 is using its referenced image 55114da9e725")
              stdout:
                  Removing image ghcr.io/security-onion-solutions/so-curator:2.3.30
----------
          ID: lasthighstate
    Function: file.touch
        Name: /opt/so/log/salt/lasthighstate
      Result: True
     Comment: Updated times on file /opt/so/log/salt/lasthighstate
     Started: 17:54:50.308671
    Duration: 5.104 ms
     Changes:
              ----------
              touched:
                  /opt/so/log/salt/lasthighstate

Summary for local
--------------
Succeeded: 625 (changed=20)
Failed:      1
--------------
Total states run:     626
Total run time:   200.193 s

**Comment by TOoSmOotH** [comment]
It should get removed next highstate

**Reply by stondino00** [reply]
If I run the highstate manually it should remove it?

**Reply by TOoSmOotH** [reply]
yes

**Reply by stondino00** [reply]
Same error when I run highstate again.

**Reply by stondino00** [reply]
Still have this on my master.  I also have one on a search node that won't delete when I run highstate more than once.

**Reply by facyber** [reply]
Hey @stondino00 

I also had same issue, but I was already root and when I run `salt-call state.highstate` I would get error "Data failed to compile: The function salt-call state.highstate is running as PID XXX....", but when I run `sudo salt-call state.highstate` as TOoSmOotH suggested, then command worked (to be honest I would expect it worked without sudo as I was already logged as root. 

Have you tried also with sudo?

EDIT: Please ignore me, now I see you did used sudo from start. My mistake.
